common/ledger/blockledger/fileledger/factory_test.go

```diff
@@ -36,16 +41,29 @@ func TestBlockStoreProviderErrors(t *testing.T) {
 	})
 
 	t.Run("open", func(t *testing.T) {
-		mockBlockStore.OpenReturns(nil, errors.New("woogie"))
+		mockBlockStoreProvider.OpenReturns(nil, errors.New("woogie"))
 		_, err := f.GetOrCreate("foo")
 		require.EqualError(t, err, "woogie")
 		require.Empty(t, f.ledgers, "Expected no new ledger is created")
 	})
 
 	t.Run("remove", func(t *testing.T) {
-		mockBlockStore.DropReturns(errors.New("oogie"))
-		err := f.Remove("foo")
-		require.EqualError(t, err, "oogie")
+		t.Run("ledger doesn't exist", func(t *testing.T) {
+			err := f.Remove("foo")
+			require.NoError(t, err)
+			require.Equal(t, 1, mockBlockStoreProvider.DropCallCount())
+		})
+
+		t.Run("dropping the blockstore fails", func(t *testing.T) {
+			mockBlockStore := &mock.BlockStore{}
```

Comment:
Better not to use BlockStore interface directly in this package. This package has already defined `FileLedgerBlockStore` local interface. Can you add Shutdown() to that interface and create mock for that interface instead?

Reply:
Sounds good. I'll try that.

Reply:
Ah, I see. I think it's just the naming of the mock that was confusing. I've updated the mock name to be consistent with the interface being mocked.